### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by JethroWarnett**
Reworked alot of the code to make it cleaner, declarative and smaller. Removed unecessary code. Fixed some bugs in undoRedo and importExportservice. Changed the css of the web app. Removed ngStyle from HTML.

### CODE REVIEW

File: src/app/components/fairChain/fairChain.component.ts
Comment (GattoPianista) [inline src/app/components/fairChain/fairChain.component.ts:90]: Should be deleted

File: src/app/components/fairChain/fairChain.component.ts
Comment (GattoPianista) [inline src/app/components/fairChain/fairChain.component.ts:284]: Maybe we could just replace a lot of this by making this.importExportService.download(this.nodes, this.edges), then it covertes it to JSON inside and also the file name  can be written inside ExportExportServie. But it is a minor thing

File: src/app/importExport.service.ts
Comment (GattoPianista) [inline src/app/importExport.service.ts]: Are we sure we can not use dataset here?

**Review by GattoPianista — APPROVED** [review]
Good work, we should fix some minor things, but you did a lot!

**Review by mdeliso97 — APPROVED** [review]
As Blendi alredy did a code review and didn't have much to add to his review (gg), I tested out the functionalities.
The versions seems to be buggy, while testing I encountered about 5 bugs:
- while changing node label, editedgemode is activated also, this might be caused by the changed condition in the onClick if statement (I solved it there previously)
- Import/export breaks the change label and color functionalities
- Edit edge label doesn't work anymore
- By editing the color of nodes, also in this case editEdgeMode is activated
- Redo/Undo also braks change color/label functionalities such like Import/Export
Maybe is better to do multiple branch bug fix for each problem before merging in dev

**Review by mdeliso97 — CHANGES_REQUESTED** [review]
Wrong approval, first we have to fix the bugs, after we can merge into dev, so please don't merge in the actual situation

**Review by Korab-Bejta — APPROVED** [review]

**Review by LYCChiang — APPROVED** [review]

**Review by GattoPianista — APPROVED** [review]

**Review by mdeliso97 — APPROVED** [review]
The corrections are good, still missin to fix the changing color/label of nodes is made also on selected edges

**Pull request merged**
